Ignore 0 failures and 0s latency in daily slack reports

taralika opened the pull request:
## Title
Ignore metrics values that are 0
cc: @msabramo 

## Relevant issues
Should fix #3598

## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🐛 Bug Fix

## Changes

<!-- List of changes -->

## Testing

<!-- Test procedure -->

## Notes

<!-- Test results -->

<!-- Points to note for the reviewer, consultation content, concerns -->

## Pre-Submission Checklist (optional but appreciated):

- [ ] I have included relevant documentation updates (stored in /docs/my-website)

## OS Tests (optional but appreciated):

- [ ] Tested on Windows
- [ ] Tested on MacOS
- [ ] Tested on Linux

taralika committed:
Ignore 0 failures and 0s latency in daily slack reports

Should fix #3598

taralika committed:
Update a comment about ignoring 0 values in addition to None

taralika committed:
Removed "5" from the string

since it's not guaranteed to always be 5, it will be at most 5, but could be less than 5 (if some values are 0s now that we ignore 0s)

krrishdholakia commented:
Is this ready for review? @taralika 

And can we have a simple unit test for this - https://github.com/BerriAI/litellm/blob/aa2f6aa5bd93260bfa7fbba864f8070bac87c424/litellm/tests/test_alerting.py#L228

taralika committed:
Added tests for ignoring 0 metrics when alerting

taralika committed:
Merge branch 'BerriAI:main' into patch-1

krrishdholakia commented:
Thanks for adding the tests @taralika - if you can share a screenshot of it passing for you, this should be good to merge in

taralika committed:
Fixed the test alert sequence

Also fixed the issue that MagicMock does not create asynchronous mocks by default.

taralika commented:
@krrishdholakia 
![image](https://github.com/BerriAI/litellm/assets/46954145/46496691-c44f-4acd-bad9-68a39832f3c3)
